File: split_sql.py
import os

with open(r'c:\Users\hugo.sales\OneDrive - EGTC INFRA S.A\Documentos\REPOSITÓRIO\onstructneon-controle-de-obras\restore_data.sql', 'r', encoding='utf-8') as f:
    lines = f.readlines()

chunks = []
current_chunk = []
for line in lines:
    current_chunk.append(line)
    if line.strip().endswith(');'):
        chunks.append("".join(current_chunk))
        current_chunk = []

if current_chunk:
    chunks.append("".join(current_chunk))

# The first chunk should include the DELETE statement
# or we can just prepend it to the first chunk.

for i, chunk in enumerate(chunks):
    file_path = fr'c:\Users\hugo.sales\OneDrive - EGTC INFRA S.A\Documentos\REPOSITÓRIO\onstructneon-controle-de-obras\restore_chunk_{i+1}.sql'
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(chunk)
    print(f"Created {file_path}")

File: split_sql.js
const fs = require('fs');
const path = require('path');

const filePath = './restore_data.sql';

const sql = fs.readFileSync(filePath, 'utf8');
const lines = sql.split('\n');

let chunk = [];
let counter = 1;

for (let i = 0; i < lines.length; i++) {
    const line = lines[i];
    chunk.push(line);
    if (line.trim().endsWith(');')) {
        const chunkPath = `./restore_chunk_${counter}.sql`;
        fs.writeFileSync(chunkPath, chunk.join('\n'), 'utf8');
        console.log(`Created ${chunkPath}`);
        chunk = [];
        counter++;
    }
}

if (chunk.length > 0) {
    const chunkPath = `./restore_chunk_${counter}.sql`;
    fs.writeFileSync(chunkPath, chunk.join('\n'), 'utf8');
    console.log(`Created final chunk: ${chunkPath}`);
}
